contrib/guix/INSTALL.md

```diff
@@ -389,10 +389,7 @@ git clone https://git.savannah.gnu.org/git/guix.git
 cd guix
 ```
 
-You will likely want to build the latest release, however, if the latest release
-when you're reading this is still 1.3.0 then you may want to use 998eda30 instead
-to avoid the issues described in [#25099](
-https://github.com/bitcoin/bitcoin/pull/25099).
+You will likely want to build the latest release.
```

Comment:
28902: I just noticed that we didn't dashify tags names here, let's add fix for it?

see 2 lines below:
```
-git branch -a -l 'origin/version-*' # check for the latest release
+git branch -a -l 'origin/v*' # check for the latest release
```